src/widgets/chart/lv_chart.c

```diff
@@ -344,34 +344,42 @@ lv_chart_series_t * lv_chart_add_series(lv_obj_t * obj, lv_color_t color, lv_cha
 
     LV_ASSERT_OBJ(obj, MY_CLASS);
 
-    lv_chart_t * chart    = (lv_chart_t *)obj;
+    lv_chart_t * chart = (lv_chart_t *)obj;
+
+    /* Allocate space for a new series and add it to the chart series linked list */
     lv_chart_series_t * ser = _lv_ll_ins_head(&chart->series_ll);
     LV_ASSERT_MALLOC(ser);
     if(ser == NULL) return NULL;
 
-    lv_coord_t def = LV_CHART_POINT_NONE;
-
-    ser->color  = color;
+    /* Allocate memory for point_cnt points, handle failure below */
     ser->y_points = lv_malloc(sizeof(lv_coord_t) * chart->point_cnt);
     LV_ASSERT_MALLOC(ser->y_points);
 
     if(chart->type == LV_CHART_TYPE_SCATTER) {
         ser->x_points = lv_malloc(sizeof(lv_coord_t) * chart->point_cnt);
         LV_ASSERT_MALLOC(ser->x_points);
+        /* NOTE: Should we return if ser->x_points is NULL? Is this an out of memory error? */
+    }
+    else {
+        ser->x_points = NULL;
     }
     if(ser->y_points == NULL) {
         _lv_ll_remove(&chart->series_ll, ser);
         lv_free(ser);
         return NULL;
     }
 
+    /* Set series properties on successful allocation */
+    ser->color = color;
     ser->start_point = 0;
     ser->y_ext_buf_assigned = false;
     ser->hidden = 0;
-    ser->x_axis_sec = axis & LV_CHART_AXIS_SECONDARY_X ? 1 : 0;
-    ser->y_axis_sec = axis & LV_CHART_AXIS_SECONDARY_Y ? 1 : 0;
+    ser->x_axis_sec = axis & LV_CHART_AXIS_SECONDARY_X;
```

Comment:
Why is this change? We want to store 1 or 0 not the result of &-ing.

Reply:
You're right, I got confused and thought it was && instead of just one, I'll revert it.